closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -1661,30 +1763,55 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_RSADP
     msg_len = (*OSSL_RSA_private_encrypt)(kLen, kNative, mNative, rsaKey, RSA_NO_PADDING);
 
     if ((-1 != verify) && (-1 != msg_len)) {
-        if (verify == kLen) {
-            k2 = malloc(kLen * (sizeof(unsigned char)));
-            if (NULL != k2) {
-
-                /* mNative is size 'verify' */
-                msg_len2 = (*OSSL_RSA_public_decrypt)(verify, mNative, k2, rsaKey, RSA_NO_PADDING);
-                if (-1 != msg_len2) {
-
-                    int i;
-                    for (i = 0; i < verify; i++) {
-                        if (kNative[i] != k2[i]) {
-                            msg_len = -2;
-                            break;
+        if (verify == msg_len) {
+            if(-1 != rsadpBNptr) {
```

Comment:
Space after `if`.
Why did you choose `-1` instead of `0` as the special value?

Reply:
Zero is slightly easier to deal with and it's the natural choice for a pointer.

Reply:
Okay, changed it to zero.